pandas/tests/indexes/timedeltas/test_timedelta_range.py

```diff
@@ -61,3 +61,33 @@ def test_errors(self):
         # too many params
         with pytest.raises(ValueError, match=msg):
             timedelta_range(start="0 days", end="5 days", periods=10, freq="H")
+
+    @pytest.mark.parametrize(
+        "start, end, freq",
+        [
+            ("1D", "10D", "2D"),
+            ("2D", "30D", "3D"),
+            ("2s", "50s", "5s"),
+            # tests that worked before GH 33498:
+            ("4D", "16D", "3D"),
+            ("8D", "16D", "40s"),
+        ],
+    )
+    def test_timedelta_range_freq_divide_end(self, start, end, freq):
+        # GH 33498 only the cases where `(end % freq) == 0` used to fail
+
+        def mock_timedelta_range(start=None, end=None, **kwargs):
+            epoch = Timestamp(0)
+            if start is not None:
+                start = epoch + Timedelta(start)
+            if end is not None:
+                end = epoch + Timedelta(end)
+            result = date_range(start=start, end=end, **kwargs) - epoch
```

Comment:
why would this be date_range?

i really don't like mocking like this, can you not hard the start, num periods for example in the parmeterization?

Reply:
> i really don't like mocking like this, can you not hard the start, num periods for example in the parmeterization?

Yeah, you are right. I just did a manual calculation instead to have the proper expected number of periods. I'll let you check the change.

> why would this be date_range?

For no good reason I guess :slightly_smiling_face:. I did a debug/test on this feature using a mocking of date_range in a notebook to compare and understand the root cause of the problem. Anyway, this is fixed :wink: